openmc/data/resonance_covariance.py

```diff
@@ -0,0 +1,841 @@
+                                        gg[j], gf[j], gx[j]])
+                    columns = ['energy', 'L', 'J', 'totalWidth', 'neutronWidth',
+                           'captureWidth', 'fissionWidth', 'competitveWidth']
+                    sample_params = pd.DataFrame.from_records(records, columns=columns)
+                    samples.append(sample_params)
+
+        # Handling RM Sampling
+        if formalism == 'rm':
+            if mpar == 3:
+                param_list = ['energy','neutronWidth','captureWidth']
+                mean_array = pd.DataFrame.as_matrix(parameters[param_list])
+                spin = pd.DataFrame.as_matrix(parameters['J'])
+                l_value = pd.DataFrame.as_matrix(parameters['L'])
+                gfa = pd.DataFrame.as_matrix(parameters['fissionWidthA'])
+                gfb = pd.DataFrame.as_matrix(parameters['fissionWidthB'])
+                mean = mean_array.flatten()
+                for i in range(n_samples):
+                    sample = np.random.multivariate_normal(mean,cov)
+                    energy = sample[0::3]
+                    gn = sample[1::3]
+                    gg = sample[2::3]
+                    records = []
+                    for j, E in enumerate(energy):
+                        records.append([energy[j], l_value[j], spin[j], gn[j],
+                                        gg[j], gfa[j], gfb[j]])
+                    columns = ['energy', 'L', 'J', 'neutronWidth',
+                           'captureWidth', 'fissionWidthA','fissionWidthB']
+                    sample_params = pd.DataFrame.from_records(records, columns=columns)
+                    samples.append(sample_params)
+    
+            elif mpar == 5:
+                param_list = ['energy','neutronWidth','captureWidth',
+                              'fissionWidthA','fissionWidthB']
+                mean_array = pd.DataFrame.as_matrix(parameters[param_list])
+                spin = pd.DataFrame.as_matrix(parameters['J'])
+                l_value = pd.DataFrame.as_matrix(parameters['L'])
+                mean = mean_array.flatten()
+                for i in range(n_samples):
+                    sample = np.random.multivariate_normal(mean,cov)
+                    energy = sample[0::5]
+                    gn = sample[1::5]
+                    gg = sample[2::5]
+                    gfa = sample[3::5]
+                    gfb = sample[4::5]
+                    records = []
+                    for j, E in enumerate(energy):
+                        records.append([energy[j], l_value[j], spin[j], gn[j],
+                                        gg[j], gfa[j], gfb[j]])
+                    columns = ['energy', 'L', 'J', 'neutronWidth',
+                           'captureWidth', 'fissionWidthA','fissionWidthB']
+                    sample_params = pd.DataFrame.from_records(records, columns=columns)
+                    samples.append(sample_params)
+    
+        self.samples = samples
```

Comment:
The docstring says that the function returns samples. Here it actually sets it as an attribute. I would have the function `return samples` rather than setting an attribute (which is not documented). You'll need to change the function signature for `reconstruct` to accept a dataframe of sampled parameters rather than an index into the list.